gui/src/app/FileEditor/StanFileEditor.tsx

```diff
@@ -22,81 +22,38 @@ type Props = {
   setEditedFileContent: (text: string) => void;
   onDeleteFile?: () => void;
   readOnly: boolean;
-  setCompiledUrl: (s: string) => void;
 };
 
-type CompileStatus = "preparing" | "compiling" | "compiled" | "failed" | "";
-
 const StanFileEditor: FunctionComponent<Props> = ({
   fileName,
   fileContent,
   onSaveContent,
   editedFileContent,
   setEditedFileContent,
   readOnly,
-  setCompiledUrl,
 }) => {
   const { stancErrors, requestFormat } = useStanc(
     "main.stan",
     editedFileContent,
     setEditedFileContent,
   );
 
+  const { setValidSyntax } = useContext(CompileContext);
+
+  const { compileStatus, compileMessage, compile } = useContext(CompileContext);
+
   const validSyntax = useMemo(() => {
     return stancErrors.errors === undefined;
   }, [stancErrors]);
 
+  useEffect(() => {
+    setValidSyntax(validSyntax);
+  }, [validSyntax, setValidSyntax]);
+
```

Comment:
[optional, not sure if it's actually a good idea] this logic could be moved inside the useStanc hook

Reply:
Can we figure this out later?

Reply:
Fine by me!